File: ProgramlamaDilleriPrensipleri/Odev2/KISIM1/Odev2Kutuphane/src/rastgelekisiuret/RastgeleKisi.java
/**
*
* @author Kader Miyanyedi kader.miyanyedi@ogr.sau.edu.tr
* 		Yasin Altunbasak yasin.altunbasak@ogr.sau.edu.tr
* @since 20.04.2020
* <p>
* Tüm bilgileri toplar ve geri bir string döndürür bu string dosyaya yazilir
* </p>
*/
package rastgelekisiuret;

import rastgelekisiuret.Telefon;
import rastgelekisiuret.Yas;

public class RastgeleKisi {
    Kisi YeniKisi = new Kisi();
    Yas YeniYas = new Yas();
    Telefon YeniTelefon = new Telefon();
    ImeiNo YeniImei = new ImeiNo();
    KimlikNo YeniKimlik = new KimlikNo();

    public String KisiOlustur() {
        String Gonder = YeniKimlik.KimlikNo() + " " + YeniKisi.KisiOlustur() + " " + YeniYas.YasOlustur()+ " " + YeniTelefon.TelefonNoUret() + " (" + YeniImei.ImeiNoUret() + ")";
        return Gonder;
    }

    public String KimlikKontrol(String Kimlik) {

        KimlikNo Tckontrol = new KimlikNo();
        return Tckontrol.KimlikNoKontrol(Kimlik);
    }

    public String ImeiKontrol(String Imei) {
        ImeiNo ImeiKontrol = new ImeiNo();
        return ImeiKontrol.ImeıKontrol(Imei);
    }
    
}

File: ProgramlamaDilleriPrensipleri/Odev2/KISIM2/src/RastgeleKisi.c
/**
*
* @author Kader Miyanyedi kader.miyanyedi@ogr.sau.edu.tr
          Yasin Altunbaşak yasin.altunbasak@ogr.sau.edu.tr
* @since 20.04.2020
* <p>
* Tüm bilgileri kisiler.txt ye yazar. Yıkıcı fonksiyonu cagrılınca tüm yıkıcı fonksiyonlar çağrılır. Kesinlikle çöp oluşmaz.
* </p>
*/
#include "../include/RastgeleKisi.h"

rskisi rskisiOlustur(){
   rskisi rs_kisi = (rskisi)malloc(sizeof(struct RASTGELEKISI));
   rs_kisi->im = imeiOlustur();
   rs_kisi->km = kimlikNoOlustur();
   rs_kisi->tf = telefonOlustur();
   rs_kisi->ys = YasOlustur(); 
   rs_kisi->kisi_ = kisiOlustur();
   return rs_kisi;

}
void rskisi_uret(rskisi rsk, int kisiSayisi){

   FILE *fp = fopen("Kisiler.txt","w");

   for(int i = 0; i<kisiSayisi; i++){
      char* imei = imeino_uret(rsk->im);
      char* kimlik = kimlikno_uret(rsk->km);
      char* isim = kisi_uret(rsk->kisi_);
      int yas = Yasuret(rsk->ys);
      char* telefon = telefonNo_uret(rsk->tf);
	   fprintf(fp,"%.11s %s %d %.11s (%.15s)\n",kimlik,isim,yas,telefon,imei);
   }

	fclose(fp);
  
}
void rskisiYoket(rskisi r){
      imeiYoket(r->im);
      kimlikNoYoket(r->km);
      kisiYoket(r->kisi_);
      YasYoket(r->ys);
      telefonYoket(r->tf);
   if(r == NULL) return;
	free(r);
	r = NULL;
}
